backend/src/schema/resolvers/notifications.spec.js

```diff
@@ -145,80 +243,201 @@ describe('given some notifications', () => {
+                        reasonDescription: 'This user is harassing me with bigoted remarks!',
+                        reportedResource: {
+                          __typename: 'User',
+                          id: 'badWomen',
+                          name: 'Mrs. Badwomen',
+                        },
+                      },
+                    ],
+                  },
+                },
+                {
+                  createdAt: '2020-01-14T12:33:48.651Z',
+                  updatedAt: '2020-01-14T12:33:48.651Z',
+                  read: false,
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnComment',
+                    filed: [
+                      {
+                        reasonCategory: 'discrimination_etc',
+                        reasonDescription: 'This user is harassing me!',
+                        reportedResource: {
+                          __typename: 'Comment',
+                          id: 'c4',
+                          content:
+                            'I am harassing content in a harassing comment to a bad post !!!',
+                        },
+                      },
+                    ],
+                  },
+                },
+                expect.objectContaining({
+                  createdAt: '2019-08-31T17:33:48.651Z',
+                  updatedAt: '2019-08-31T17:33:48.651Z',
+                  read: false,
+                  from: {
+                    __typename: 'Post',
+                    content: 'You have been mentioned in a post',
+                  },
+                }),
+                expect.objectContaining({
+                  createdAt: '2019-08-30T19:33:48.651Z',
+                  updatedAt: '2019-08-30T19:33:48.651Z',
+                  read: false,
+                  from: {
+                    __typename: 'Comment',
+                    content: 'You have been mentioned in a comment',
+                  },
+                }),
+                expect.objectContaining({
+                  createdAt: '2019-08-30T17:33:48.651Z',
+                  updatedAt: '2019-08-30T17:33:48.651Z',
+                  read: true,
+                  from: {
+                    __typename: 'Post',
+                    content: 'Already seen post mention',
+                  },
+                }),
+                expect.objectContaining({
+                  createdAt: '2019-08-30T15:33:48.651Z',
+                  updatedAt: '2019-08-30T15:33:48.651Z',
+                  read: true,
+                  from: {
+                    __typename: 'Comment',
+                    content: 'You have seen this comment mentioning already',
+                  },
+                }),
+              ],
             },
             errors: undefined,
-          })
+          }
+
+          const response = await query({ query: notificationQuery, variables })
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-23T19:00:04Z" title="Thursday, January 23rd 2020, 8:00:04 pm +01:00">Jan 23, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -145,80 +233,188 @@ describe('given some notifications', () => {
 
       describe('no filters', () => {
         it('returns all notifications of current user', async () => {
-          const expected = [
-            {
-              from: {
-                __typename: 'Comment',
-                content: 'You have seen this comment mentioning already',
-              },
-              read: true,
-              createdAt: '2019-08-30T15:33:48.651Z',
-            },
-            {
-              from: {
-                __typename: 'Post',
-                content: 'Already seen post mention',
-              },
-              read: true,
-              createdAt: '2019-08-30T17:33:48.651Z',
-            },
-            {
-              from: {
-                __typename: 'Comment',
-                content: 'You have been mentioned in a comment',
-              },
-              read: false,
-              createdAt: '2019-08-30T19:33:48.651Z',
-            },
-            {
-              from: {
-                __typename: 'Post',
-                content: 'You have been mentioned in a post',
-              },
-              read: false,
-              createdAt: '2019-08-31T17:33:48.651Z',
-            },
-          ]
-
-          await expect(query({ query: notificationQuery, variables })).resolves.toMatchObject({
+          const expected = {
             data: {
-              notifications: expect.arrayContaining(expected),
+              notifications: expect.arrayContaining([
+                expect.objectContaining({
+                  from: {
+                    __typename: 'Comment',
+                    content: 'You have seen this comment mentioning already',
+                  },
+                  read: true,
+                  createdAt: '2019-08-30T15:33:48.651Z',
+                }),
+                expect.objectContaining({
+                  from: {
+                    __typename: 'Post',
+                    content: 'Already seen post mention',
+                  },
+                  read: true,
+                  createdAt: '2019-08-30T17:33:48.651Z',
+                }),
+                expect.objectContaining({
+                  from: {
+                    __typename: 'Comment',
+                    content: 'You have been mentioned in a comment',
+                  },
+                  read: false,
+                  createdAt: '2019-08-30T19:33:48.651Z',
+                }),
+                expect.objectContaining({
+                  from: {
+                    __typename: 'Post',
+                    content: 'You have been mentioned in a post',
+                  },
+                  read: false,
+                  createdAt: '2019-08-31T17:33:48.651Z',
+                }),
+                {
+                  createdAt: '2020-01-15T16:33:48.651Z',
+                  read: false,
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnUser',
+                    filed: [
+                      {
+                        reasonCategory: 'discrimination_etc',
+                        reasonDescription: 'This user is harassing me with bigoted remarks!',
+                        reportedResource: {
+                          __typename: 'User',
+                          id: 'badWomen',
+                          name: 'Mrs. Badwomen',
+                        },
+                      },
+                    ],
+                  },
+                },
+                {
+                  createdAt: '2020-01-16T10:33:48.651Z',
+                  read: true,
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnPost',
+                    filed: [
+                      {
+                        reasonCategory: 'other',
+                        reasonDescription:
+                          "This shouldn't be shown to anybody else! It's my private thing!",
+                        reportedResource: {
+                          __typename: 'Post',
+                          id: 'p4',
+                          title: 'Bad Post',
+                          content: 'I am bad content !!!',
+                        },
+                      },
+                    ],
+                  },
+                },
+                {
+                  createdAt: '2020-01-14T12:33:48.651Z',
+                  read: false,
+                  reason: 'filed_report_on_resource',
+                  from: {
+                    __typename: 'Report',
+                    id: 'reportOnComment',
+                    filed: [
+                      {
+                        reasonCategory: 'discrimination_etc',
+                        reasonDescription: 'This user is harassing me!',
+                        reportedResource: {
+                          __typename: 'Comment',
+                          id: 'c4',
+                          content: 'I am bad content in a bad comment to a bad post !!!',
+                        },
+                      },
+                    ],
+                  },
+                },
+              ]),
             },
             errors: undefined,
-          })
+          }
+
+          const response = await query({ query: notificationQuery, variables })
```


```suggestion
          const { data: { notifications } } = await query({ query: notificationQuery, variables })
```

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-01-24T11:45:11Z" title="Friday, January 24th 2020, 12:45:11 pm +01:00">Jan 24, 2020</time>_

---


I don’t like to do this, because I got to the –  in my eyes – useful habit to have the `errors: undefined,` within the object that I compare.
Then the next coder can explicitly see that it’s in the response and not ending up with the same problem I had: There is an undefined error. What I did wrong.
And spend a lot of time to try to find out …